src/main/scala/tsiHost/TsiHostWidgetUnitTest.scala

```diff
@@ -0,0 +1,435 @@
+ */
+class TSIHostWidgetTest(implicit p: Parameters) extends LazyModule {
+  // params matching the configuration for the TSIHost widget
+  val systemBeatBytes = 8 // should match with the host beat bytes
+  val params = p(PeripheryTSIHostKey).head
+  val targetLineBytes = params.serdesParams.managerPortParams.maxTransfer // should match with host manager line sz
+                                                                                      // note: this is hardcoded in the default to 64B
+  val hostAddrSet = params.serdesParams.managerPortParams.managers(0).address(0) // should match with target manager addr set
+  //val targetAddrSet = params.serdesParams.managerParams.address(0) // should match with host manager addr set
+  val targetNumXacts = params.serdesParams.clientPortParams.clients(0).sourceId.end // should match the host client num Xacts
+
+  // TSIHost widget in host-land connecting to the target Serdes
+  val hostTSIHostWidget = LazyModule(new TLTSIHostWidget(systemBeatBytes, params))
+
+  // lives in target-land (connects to the TSIHost widget in host-land)
+  val targetSerdes = LazyModule(new TLSerdesser(
+    w = params.serialIfWidth,
+    clientPortParams = params.serdesParams.clientPortParams,
+    managerPortParams = params.serdesParams.managerPortParams))
+
+  targetSerdes.managerNode := TLBuffer() := targetSerdes.clientNode
+
+  // ram living in host-land. for the fuzzer to read and write from
+  val hostRam = LazyModule(new TLTestRAM(
+    address = hostAddrSet,
+    beatBytes = systemBeatBytes))
+
+  // connect the host node to the host rom
+  hostRam.node := TLFragmenter(systemBeatBytes, targetLineBytes) := TLBuffer() := hostTSIHostWidget.externalClientNode
+
+  // amount of words to fuzz (bounded by hostAddrSet.max)
+  val totalWordsInput = 50
+  require(hostAddrSet.max >= totalWordsInput)
+  val (inMMIOWriteSeq, outMMIOReadSeq) = TSIFuzzerGeneratorWriteReadSeq(serialIfWidth = SerialAdapter.SERIAL_TSI_WIDTH,
+                                                                        totalWords = totalWordsInput,
+                                                                        maxReqWords = 5,
+                                                                        baseAddress = BigInt(0x0))
+
+  // convert to write, and read expects with the data to patterns to send to pattern pusher
+  val      writePatternSeq = inMMIOWriteSeq.map(WritePattern(params.mmioBaseAddress + TSIHostWidgetCtrlRegs.txQueueOffset,
+                                                             2,
+                                                             _))
+  // get rid of the MSB (fifo full status)
+  val readExpectPatternSeq = outMMIOReadSeq.map(data => ReadExpectPattern(params.mmioBaseAddress + TSIHostWidgetCtrlRegs.rxQueueOffset,
+                                                                          2,
+                                                                          (data << 1) >> 1))
```

Comment:
Does this work? I thought the left shift would add a bit.

Reply:
Actually, I think this does nothing since the data is 32b to begin with. I'll remove this logic since the issue is actually  that you can't deal with non-blocking queues in this unit test